pkg/agent/proxy/proxier.go

```diff
@@ -56,9 +57,12 @@ type proxier struct {
 	serviceInstalledMap k8sproxy.ServiceMap
 	// endpointsMap stores endpoints we expect to be installed.
 	endpointsMap types.EndpointsMap
-	// endpointInstalledMap stores endpoints we actually installed.
-	endpointInstalledMap map[k8sproxy.ServicePortName]map[string]struct{}
-	groupCounter         types.GroupCounter
+	// endpointsInstalledMap stores endpoints we actually installed.
+	endpointsInstalledMap types.EndpointsMap
+	// endpointReferenceCounter stores the number of times an Endpoint is referenced by Services.
+	endpointReferenceCounter map[string]uint
```

Comment:
it may just be me, but I prefer using signed integers in this case, because if there is a bug in the code and the number becomes negative it's much easier to spot.